### PR TITLE
Feature/build inst su links

### PR DESCRIPTION
**Pull request by philratcliffe**
### What

Add links and names for an institutions student unions (based on location) to the institution resource. The algorithm is described below. Note: this is an update to the original mapping. Discussed and agreed with Nathan.

1. Build a location lookup table from the location entries in the KIS dataset. The key into the lookup tables comprises: UKPRN and LOCID. If the LOCATION.LOCUKPRN is present it is used for the UKPRN part otherwise use LOCATION.UKPRN is used. 

2. Iterate through all the courses for an institution and for each COURSELOCATION extract the LOCID (if present) and combine with INSTITUTION.PUBUKPRN to enable a location lookup key to be created. Lookup the location and build the student union link.

Also added link for institution_homepage to institution resource. 

Refactor and reorganise some of the code.

### How to review

Review the code

Run the CreateInstBlobTrigger function with the latest KIS dataset and check the documents created are as expected.

### CODE REVIEW

File: CreateInstBlobTrigger/locations.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+            UKPRN
+            LOCID
+
+        If LOCATION.LOCUKPRN is present, we use that for the UKPRN part
+        otherwise we use LOCATION.UKPRN for the UKPRN part.
+
+        The LOCID part is the LOCATION.LOCID
+
+        We are not sure if this mapping is correct; it may need a
```
Comment (nshumoogum) [inline CreateInstBlobTrigger/locations.py]: Surely this comment can now be removed?

File: SharedCode/utils.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 """Functions shared by Azure Functions"""
 
 import os
+import html
```
Comment (nshumoogum) [inline SharedCode/utils.py]: Is this needed?

File: CreateInstBlobTrigger/locations.py (hunk 1 above)
Comment (philratcliffe) [reply CreateInstBlobTrigger/locations.py]: Yes, confirmed with Catherine this morning this alg okay. 👍

File: SharedCode/utils.py (hunk 2 above)
Comment (philratcliffe) [reply SharedCode/utils.py]: Yes, html.unescape used in get_english_welsh_item to unescape HTML entries in string such  <LOCNAME>Abingdon &amp;amp; Witney College (Common Leys Campus)</LOCNAME>

**Pull request merged**